portfolio/src/main/webapp/scripts/generateCircles.js

```diff
@@ -0,0 +1,80 @@
+const GOOGLE_COLORS = ["#f4c20d", "#4885ed", "#3cba54"] // yellow, blue, green
+const ICONS = ["contact", "code", "education", "me", "work", "achievement"];
+
+let contents = [];
+
+const FULL_CIRCLE_DEGREE = 360;
+const SATELLITE_NUMBER = 6;
+const RADIUS = 250;
+const SATELLITE_RADIUS = 90;
+
+const parentdiv = document.getElementById('parentdiv');
+
+const convertCoordinateToString = (coor) => {
+  const offsetToParentCenter = parseInt(parentdiv.offsetWidth / 2); //assumes parent is square
```

Comment:
This assumption doesn't seem necessary.  You can treat a rectangular container as an ellipse and use (offsetWidth, offsetHeight) to get your x and y radii, then calculate x and y based on that.

If you are going to include this assumption, you should validate it, and handle the case where it is not satisfied.  Even if you just crash, that is better than providing an unpredictable user experience.